src/main/java/games/strategy/util/memento/PropertyBagMemento.java

```diff
@@ -0,0 +1,103 @@
+package games.strategy.util.memento;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implementation of {@link Memento} that represents the originator state as a property bag.
```

Comment:
'as a property bag' is a recursive definition ('PropertyBagMemento').  Replacing the words 'property bag' with a definition for them would likely help.

Also would be interesting to have noted any limitations of what can be property bagged and what cannot. Notes on alternative choices would be interesting to, for example when would we really want to use a `PropertyBagMemento` when we want a `Memento`, and when is it not appropriate?